src/coreclr/jit/assertionprop.cpp

```diff
@@ -3821,14 +3840,92 @@ GenTree* Compiler::optAssertionProp_LclFld(ASSERT_VALARG_TP assertions, GenTreeL
 GenTree* Compiler::optAssertionProp_Asg(ASSERT_VALARG_TP assertions, GenTreeOp* asg, Statement* stmt)
 {
     GenTree* rhs = asg->gtGetOp2();
-    if (asg->OperIsCopyBlkOp() && varTypeIsStruct(rhs))
```

Comment:
Why the `varTypeIsStruct(rhs)` condition removal, along with the assert in `optZeroObjAssertionProp`?

Reply:
Because `optZeroObjAssertionProp` was extended to handle LCL_FLD.